vowpalwabbit/core/src/reductions/explore_eval.cc

```diff
@@ -164,6 +165,8 @@ void do_actual_learning(explore_eval& data, multi_learner& base, VW::multi_ex& e
       if (data.known_cost.action == a_s[i].action) { action_probability = a_s[i].score; }
     }
 
+    if (action_probability == 0) { return; }
```

Comment:
```suggestion
    if (action_probability == 0.f) { return; }
```

Also, should this be a comparison with some tolerance instead of exact?

Reply:
if it is LAS then the action will be missing (action_mask is after explore_eval) and the for loop won't assign anything and so the action_probability will be as it was set on line 162

otherwise we go to previous explore_eval and it might be small but not zero so don't think we should skip in that case